### PR TITLE
Refactor return value typings for container clientAPI

### PR DESCRIPTION
**Pull request by ndricimrr**
**Description**

Changes proposed in this pull request:
- **isSpecialMf** renamed to **isCompoundChild**
- Add missing **JSdoc** to functions
- Adjust **tests**
- Add **Typescript** return val typings
- Remove getAttributes usage in favor of Svelte > thisComponent.attribute

**Related issue(s)**

### CODE REVIEW

File: container/src/services/webcomponents.service.ts [hunk 1]
```diff
@@ -127,54 +139,59 @@ export class WebComponentService {
...
           return {};
         }
-        let result = this.thisComponent.getAttribute('node-params') || {};
-        result = JSON.parse(result);
         if (shouldDesanitise) {
-          return deSanitizeParamsMap(result);
+          return deSanitizeParamsMap(this.thisComponent.nodeParams)
         }
-        return result;
+        return this.thisComponent.nodeParams || {} || undefined;
```
Comment (JohannesDoberer) [inline container/src/services/webcomponents.service.ts:154]:
```suggestion
        return this.thisComponent.nodeParams || {};

```

**Review by JohannesDoberer — APPROVED** [review]
One small thing besides that
LGTM 👍 !

File: container/src/services/webcomponents.service.ts [hunk 2]
```diff
@@ -97,16 +109,16 @@ export class WebComponentService {
               });
             });
           },
-          getCurrentTheme: () => {
-            return this.thisComponent.getAttribute('theme');
```
Comment (ndricimrr) [inline container/src/services/webcomponents.service.ts:101]: leave it theme| undefined

File: container/src/services/webcomponents.service.ts [hunk 3]
```diff
@@ -97,16 +109,16 @@ export class WebComponentService {
...
+          getCurrentTheme: () : string | undefined | boolean => {
+            return this.thisComponent.theme || false || 'defaultTheme';
           }
         };
       },
-      getCurrentLocale: () => {
-        return this.thisComponent.getAttribute('locale');
+      getCurrentLocale: () : string =>  {
+        return this.thisComponent.locale || 'en';
```
Comment (ndricimrr) [inline container/src/services/webcomponents.service.ts:118]: leave as theme

File: container/src/services/webcomponents.service.ts [hunk 4]
```diff
@@ -127,54 +139,59 @@ export class WebComponentService {
...
-        let result = this.thisComponent.getAttribute('client-permissions') || {};
-        result = JSON.parse(result);
-        return result;
+      getClientPermissions: (): Object  => {
+        return this.thisComponent.clientPermissions || {};
       },
-      getUserSettings: () => {
-        return JSON.parse(this.thisComponent.getAttribute('user-settings')) || {};
+      getUserSettings: (): Object | null => {
+        return this.thisComponent.userSettings || null;
```
Comment (ndricimrr) [inline container/src/services/webcomponents.service.ts:175]: leave it as it was.

**Pull request merged**